lib/service/supervisor.go

```diff
@@ -126,10 +130,14 @@ type LocalSupervisor struct {
 	closeContext context.Context
 	signalClose  context.CancelFunc
 
-	// exitContext is closed when someone emits Exit event
+	// exitContext is closed when someone emits a hard Exit event
 	exitContext context.Context
 	signalExit  context.CancelFunc
 
+	// shutdownContext is closed when someone emits a graceful or hard Exit event
+	shutdownContext context.Context
```

Comment:
Rename ctx and cancel similarly as above (assuming you took the suggestion above)?